src/API/ProductTemplates.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * REST API Admin Product Templates controller
+ *
+ * Handles requests to the admin templates endpoint.
+ */
+
+namespace Automattic\WooCommerce\Admin\API;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Admin Notes controller class.
+ *
+ * @extends WC_REST_CRUD_Controller
+ */
+class ProductTemplates extends \WC_REST_Products_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc-analytics';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'products/templates';
+
+	/**
+	 * Register the routes for admin notes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => array_merge(
+						$this->get_endpoint_args_for_item_schema( \WP_REST_Server::CREATABLE ),
+						array(
+							'template_name' => array(
+								'required'    => true,
+								'type'        => 'string',
+								'description' => __( 'Product template name.', 'woocommerce-admin' ),
+							),
+						)
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Get a single note.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function create_item( $request ) {
```

Comment:
Seeing that this function is largely a copy-paste of [`OnboardingTasks::import_sample_product()`](https://github.com/woocommerce/woocommerce-admin/blob/b49c96ee2201445c9cb32a5cdf324c33fd1a3a06/src%2FAPI%2FOnboardingTasks.php#L126), can we refactor that into a reusable method for both endpoints?

And while we're reusing that code.. and this feature is part of the onboarding.. why not have this endpoint code live in `OnboardingTasks.php`?